### PR TITLE
Add script to generate JSON file with list of compiler versions

### PR DESCRIPTION
**Pull request by mds1**
Adds an `update` script that can be ran with `node update` to generate the `list.json` file, along with the other outputs shown here. The `update` script is based on the [upstream version](https://github.com/ethereum/solc-bin/blob/0b238f266bdb099703628410f7df4fd17ab72fcd/update), with changes:
- d43a1a3 adds error handling to account for things like invalid semver versions such as `v0.8.4-broken_alpha`
- 2037c2b handles the fact that OVM solc versioning is (seemingly) different from ordinary solc versioning in two ways:
    - `longVersion`s do not have a commit hash by default in this repo
    - Short version numbers are not unique, e.g. there are two v0.7.6 versions

To update the hardhat-etherscan plugin to support optimism verification, it would help if `list.json` is published somewhere so hardhat-etherscan can fetch it, as it does for solc [here](https://github.com/nomiclabs/hardhat/blob/c9e8a609be8b7751ce375a22159d4d2af162dad2/packages/hardhat-etherscan/src/solc/version.ts). (We should be able to this by reading from hardhat config if required)

This PR does not setup CI to automate the process of running `node update` and pushing the build artifacts

### CODE REVIEW

**Comment by mds1** [comment]
Closing this PR as part of some general github triage of PRs I created or am assigned to that are now stale. I believe this whole repo is now outdated thanks to EVM equivalence so this can be safely closed, but feel free to reopen if that's not the case
